CMakeLists.txt

```diff
@@ -12,6 +12,13 @@ if(NOT foonathan_memory_FOUND)
   # unless the library was explicitly added as a static library.
   option(BUILD_SHARED_LIBS "Create shared libraries by default" ON)
 
+  # Option to build memory examples
+  option(BUILD_MEMORY_EXAMPLES "Build memory examples" OFF)
+  # Option to build memory tests
+  option(BUILD_MEMORY_TESTS "Build memory tests" OFF)
+  # Option to build memory tools
+  option(BUILD_MEMORY_TOOLS "Build memory tools" ON)
```

Comment:
Examples and Tests require tool to be built.  `BUILD_MEMORY_TOOLS` should be set when either `BUILD_MEMORY_EXAMPLES` or `BUILD_MEMORY_TESTS` is `ON`.